src/main/java/de/berlin/lostberlin/controller/OrderController.java

```diff
@@ -1,85 +1,106 @@
+        return orderService.retrieveOrderByOrderNr(orderNr);
     }
 
-    @PostMapping("/orders")
+    @PostMapping
+    public ResponseEntity createOrder(@Valid @RequestBody OrderPostDto orderProfile) {
 
-    public String createOrder(@Valid @RequestBody Order order){
-        if (order.getChosenBusinessIds().length==0
-                ||order.getName().equals("")
-                ||!EmailValidator.getInstance().isValid(order.getEmail())
-        ){
+        Order result = orderService.saveOrderProfile(orderProfile);
 
-            return "Bad request";
-        }
-        Order result = orderRepo.save(order);
         if (result != null) {
             try {
                 mailService.sendConfirmationMail(result);
             } catch (Exception e) {
-                return e.getMessage();
+                e.getMessage();
             }
         } else {
-            return "Failed to save order " + order.getOrderNr();
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body("Failed to save order for" + orderProfile.getName());
         }
-        return "Success";
-    }
-
-    @GetMapping("/orders/{order_number}")
-    public Order getByOrderNr(@PathVariable(value = "order_number") String orderNr) {
-        return orderRepo.findById(orderNr)
-                .orElseThrow(() -> new ResourceNotFoundException("Order", "order_number", orderNr));
+        return ResponseEntity.status(HttpStatus.CREATED).body("order created");
     }
 
-    @PutMapping("/orders/{order_number}/status")
-
-    public Order updateOrderStatus(@PathVariable (value="order_number") String orderNr, @Valid @RequestBody Order orderProfile){
-    Order order = orderRepo.findById(orderNr)
-            .orElseThrow(() -> new ResourceNotFoundException("Order", "order_number", orderNr));
-    order.setBusinessId(orderProfile.getBusinessId());
-    order.setStatus(orderProfile.getStatus());
-
-
-        Order result = orderRepo.save(order);
-
-        if(result !=null && result.getStatus().equals("confirmed")){
+    @PutMapping("/{order_number}/confirmation")
+    public ResponseEntity confirmOrder(
+            @PathVariable(value = "order_number") String orderNr,
+            @Valid @RequestParam String status,
+            @Valid @RequestParam (value = "business_id") Long businessId
+    ) {
+        Order order = orderRepo.findById(orderNr)
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, "order not found"));
+        if (order.getStatus().equals(StatusTypes.PENDING)) {
+            orderService.updateOrderStatus(orderNr, status, businessId);
+            OrderFullDao result = orderService.retrieveFullOrderProfile(orderNr, status, businessId);
+            if (result != null) {
             try {
                 mailService.sendNotificationMail(result);
             } catch (Exception e) {
                 e.getMessage();
```

Comment:
it should return error ?

Reply:
oh no, it should throw it of course. Actually, this branch wasn't supposed to be merged. I've pushed it, so that Sergey could get my current version of code in order to look for some other solutions. I'll delete this PR now. The kosher code  is in the next PR: order_service_and_controller_refactoring_and_other_fixes